api/user.go

```diff
@@ -117,6 +135,41 @@ func (ua *UserAPI) Put() { //currently only for toggle admin, so no request body
 	dao.ToggleUserAdminRole(userQuery)
 }
 
+// Post ...
+func (ua *UserAPI) Post() {
+	var req userReq
+	ua.DecodeJSONReq(&req)
+
+	err := validateUserReq(req)
+	if err != nil {
+		log.Errorf("Invalid user request, error: %v", err)
+		ua.RenderError(http.StatusBadRequest, "Invalid request for creating user")
+		return
+	}
+
+	user := models.User{Username: req.UserName, Email: req.Email, Realname: req.Realname, Password: req.Password, Comment: req.Comment}
+	exist, err := dao.UserExists(user, "email")
+	if err != nil {
+		log.Errorf("Error occurred in UserExists: %v", err)
+	}
+	if exist {
+		ua.RenderError(http.StatusConflict, "")
+		return
+	}
+
+	userID, err := dao.Register(user)
+	if err != nil {
+		log.Errorf("Error occurred in Register: %v", err)
+		ua.RenderError(http.StatusInternalServerError, "Internal error.")
+		return
+	}
+	if userID == 0 {
```

Comment:
Because the dao.Register will return 0 when encounter any error. And I will remove the valid request step as it's duplicate with validate(user models.User) under dao/register.go